pkg/apis/resiliency/v1alpha1/types.go

```diff
@@ -46,7 +46,7 @@ type Retry struct {
 	Policy      string `json:"policy,omitempty" yaml:"policy,omitempty"`
 	Duration    string `json:"duration,omitempty" yaml:"duration,omitempty"`
 	MaxInterval string `json:"maxInterval,omitempty" yaml:"maxInterval,omitempty"`
-	MaxRetries  int    `json:"maxRetries,omitempty" yaml:"maxRetries,omitempty"`
+	MaxRetries  *int   `json:"maxRetries,omitempty" yaml:"maxRetries,omitempty"`
```

Comment:
By making this a pointer, when it's serialized to JSON, a value of `0` is preserved.

See #5530 for an explanation of why this was causing errors.